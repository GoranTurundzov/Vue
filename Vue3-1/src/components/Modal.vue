<template>
    <div @click.self="closeModal" class="modal out">
        <div class="modal-dialog modal-dialog-centered">
            <div class="modal-content">
                <div class="block-header bg-primary-darker">
                    <ul class="block-options">
                        <li>
                            <button type="button" @click="closeModal" data-testid="x-button">
                                <i class="fa fa-times"></i>
                            </button>
                        </li>
                    </ul>
                    <slot name="header"></slot>
                </div>
                <div class="block-content">
                    <slot></slot>
                </div>
                <div class="modal-footer" v-if="footerSlotPassed">
                    <slot name="footer"></slot>
                </div>
            </div>
        </div>
    </div>
</template>

<script>
export default {
    name: 'Modal',
    emits: ['close-modal'],
    computed: {
        footerSlotPassed() {
            return this.$slots['footer'];
        }
    },
    methods: {
        closeModal() {
            this.$emit('close-modal');
        }
    }
};
</script>

<style scoped>
.out {
    background-color: rgba(7, 7, 7, 0.7);
    display: block;
    z-index: 2;
}
.modal-dialog {
    max-height: 92vh;
    overflow: scroll;
    -ms-overflow-style: none;
    scrollbar-width: none;
}
.modal-dialog::-webkit-scrollbar {
    display: none;
}
</style>
