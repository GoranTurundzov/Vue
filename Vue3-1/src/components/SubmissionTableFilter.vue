<template>
    <div class="form-div">
        <form class="form-horizontal">
            <div class="block-content">
                <div class="form-group">
                    <label class="col-xs-12"> Form </label>
                    <div class="col-xs-12 push-20">
                        <select
                            class="form-control"
                            v-model="form"
                            @change="emitFilters"
                            data-testid="form-name-selector"
                        >
                            <option>All</option>
                            <option v-for="form in formSelectOptions" :key="form.name">
                                {{ form.name }}
                            </option>
                        </select>
                    </div>
                </div>
                <div class="form-group">
                    <label class="col-xs-12"> Form Type </label>
                    <div class="col-xs-12 push-20">
                        <select
                            class="form-control"
                            v-model="form_type"
                            @change="emitFilters"
                            data-testid="form-selector"
                        >
                            <option>All</option>
                            <option>Popup</option>
                            <option>Page</option>
                            <option>Email</option>
                        </select>
                    </div>
                </div>
                <div class="form-group">
                    <label class="col-xs-12"> Recommendation </label>
                    <div class="col-xs-12 push-20">
                        <select
                            class="form-control"
                            v-model="recommendation"
                            @change="emitFilters"
                            data-testid="recommendation-selector"
                        >
                            <option>All</option>
                            <option>Yes</option>
                            <option>No</option>
                            <option>Maybe</option>
                        </select>
                    </div>
                </div>
                <div class="form-group">
                    <label class="col-xs-12"> Date </label>
                    <div class="col-xs-12 push-20">
                        <select
                            class="form-control"
                            v-model="date"
                            @change="emitFilters"
                            data-testid="date-selector"
                        >
                            <option>All</option>
                            <option>Today</option>
                            <option>Yesterday</option>
                            <option>Month to date</option>
                            <option>Last month</option>
                            <option>Year to date</option>
                            <option>Last year</option>
                        </select>
                    </div>
                </div>
                <div class="form-group">
                    <label class="col-xs-12"> Status </label>
                    <div class="col-xs-12 push-20">
                        <select
                            class="form-control"
                            v-model="status"
                            @change="emitFilters"
                            data-testid="status-selector"
                        >
                            <option>All</option>
                            <option>Pending</option>
                            <option>Approved</option>
                            <option>Rejected</option>
                        </select>
                    </div>
                </div>
            </div>
        </form>
    </div>
</template>

<script>
export default {
    name: 'SubmissionTableFilter',
    props: {
        propForm: {
            type: String,
            required: true,
            description: 'form name'
        },
        propFormType: {
            type: String,
            required: true,
            description: 'Filter prop'
        },
        propStatus: {
            type: String,
            required: true,
            description: 'Filter prop'
        },
        propDate: {
            type: String,
            required: true,
            description: 'Filter prop'
        },
        propRecommendation: {
            type: String,
            required: true,
            description: 'Filter prop'
        },
        formSelectOptions: {
            type: Array,
            required: true,
            description: 'Forms to select from'
        }
    },
    data() {
        return {
            form: 'All',
            form_type: 'All',
            status: 'All',
            date: 'All',
            recommendation: 'All'
        };
    },
    created() {
        this.form = this.propForm;
        this.form_type = this.propFormType;
        this.status = this.propStatus;
        this.recommendation = this.propRecommendation;
        this.date = this.propDate;
    },
    methods: {
        emitFilters() {
            this.$emit('change-filter', {
                form: this.form,
                status: this.status,
                date: this.date,
                recommendation: this.recommendation,
                form_type: this.form_type
            });
        }
    }
};
</script>
