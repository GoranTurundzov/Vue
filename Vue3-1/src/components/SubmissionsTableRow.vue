<template>
    <div class="content justify-center">
        <submissions-table-cell
            v-for="(key, value) in rowData"
            :title="value"
            :align-in-rows="alignInRows"
            :text="key"
            :key="key"
        ></submissions-table-cell>
        <div class="column" v-if="!alignInRows">
            <div>Action</div>
            <div>
                <button class="btn btn-gray btn-sm push-5-r" @click="editSubmission">
                    <i class="fa fa-edit"></i>
                </button>
                <button
                    class="btn btn-gray btn-sm push-5-r"
                    @click="activateModal"
                    data-testid="modal-activator"
                >
                    <i class="fa fa-eye"></i>
                </button>
                <button
                    class="btn btn-gray btn-sm push-5-r"
                    @click="deleteSubmission"
                    data-testid="delete-activator"
                >
                    <i class="fa fa-trash"></i>
                </button>
            </div>
        </div>
    </div>
</template>

<script>
import SubmissionsTableCell from './SubmissionsTableCell';

export default {
    components: {
        SubmissionsTableCell
    },
    name: 'SubmissionsTableRow',
    props: {
        rowData: {
            type: Object,
            required: true,
            description: 'A single row worth of data'
        },
        alignInRows: {
            type: Boolean,
            default: false,
            description: 'Direction on how the row will be printed (column =false; row = true)'
        }
    },
    emits: ['activate-modal', 'delete-submission', 'edit-submission'],
    methods: {
        activateModal() {
            this.$emit('activate-modal', this.rowData);
        },
        deleteSubmission() {
            this.$emit('delete-submission', this.rowData);
        },
        editSubmission() {
            this.$emit('edit-submission', this.rowData);
        }
    }
};
</script>

<style scoped>
.column {
    display: table-cell;
    text-align: left;
    min-width: 10.5vw;
    max-width: 10.5vw;
    font-size: 1.2rem;
    vertical-align: middle;
}
button:hover {
    transform: scale(1.2);
}
@media screen and (max-width: 1200px) {
    .content {
        flex-direction: column;
        padding: 5%;
        font-size: 1rem;
    }
    .column {
        display: block;
        width: 100%;
        min-width: 100%;
    }
}
</style>
