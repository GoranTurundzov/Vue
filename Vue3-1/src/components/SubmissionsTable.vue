<template>
    <div class="block-content">
        <div class="block">
            <select class="btn btn-default dropdown-toggle" @change="postsPerPage">
                <option value="5" selected>5</option>
                <option value="10">10</option>
                <option value="15">15</option>
                <option value="20">20</option>
            </select>
            <button
                class="btn btn-secondary margin-left"
                @click="toggleFilter"
                data-testid="open-filters-button"
            >
                Filter submissions
            </button>
            <button
                class="btn btn-primary pull-right"
                @click="createNewSubmission"
                data-testid="create-submission-button"
            >
                Create submission
            </button>
        </div>
        <div class="block block-bordered block-rounded">
            <submissions-table-row
                v-for="(item, index) in loadedSubmissions"
                :key="index"
                :row-data="item"
                class="row-hover"
                @activate-modal="triggerModal"
                @delete-submission="deleteSubmission"
                @edit-submission="editSubmission"
            ></submissions-table-row>
        </div>
        <div class="row justify-content-center">
            <button
                class="btn btn-primary justify-content-center"
                data-testid="test-load-more"
                v-if="showButton"
                @click="loadMore"
            >
                Load More
            </button>
        </div>
    </div>
    <modal v-if="modalActive" name="previev-submission-modal" @close-modal="triggerModal">
        <template v-slot:header>
            <h3>{{ modalData.name }}'s Submission</h3>
        </template>
        <submissions-table-row :align-in-rows="true" :row-data="modalData"></submissions-table-row>
    </modal>
    <modal v-if="filterModal" @close-modal="toggleFilter" name="filter-modal">
        <template v-slot:header>
            <h3>Filter Submissions</h3>
        </template>
        <submission-table-filter
            :prop-form-type="appliedFilters.form_type"
            :prop-status="appliedFilters.status"
            :prop-date="appliedFilters.date"
            :prop-recommendation="appliedFilters.recommendation"
            :prop-form="appliedFilters.form"
            :form-select-options="formSelectOptions"
            @cancel-filters="toggleFilter"
            @change-filter="stockFilter"
        >
        </submission-table-filter>
        <template v-slot:footer>
            <button class="btn btn-seconday" @click="toggleFilter">
                Cancel
            </button>
            <button
                class="btn btn-primary"
                @click="submitFilter"
                data-testid="submit-filters-button"
            >
                Continue
            </button>
        </template>
    </modal>
</template>

<script>
import SubmissionsTableRow from '@/components/SubmissionsTableRow';
import Modal from '@/components/Modal';
import SubmissionTableFilter from '@/components/SubmissionTableFilter';
import filterData from '../services/filterData';

export default {
    name: 'SubmissionsTable',
    components: {
        SubmissionsTableRow,
        Modal,
        SubmissionTableFilter
    },
    props: {
        tableData: {
            type: Array,
            required: true,
            description: 'it provides data for each row of the table'
        },
        formSelectOptions: {
            type: Array,
            required: true,
            description: 'Forms to select from'
        }
    },
    emits: ['delete-this-submission', 'create-new-submission', 'edit-this-submission'],
    data() {
        return {
            rowsToLoad: 5,
            timesLoaded: 1,
            filterModal: false,
            modalActive: false,
            modalData: null,
            selectedFilters: null,
            appliedFilters: {
                form: 'All',
                form_type: 'All',
                recommendation: 'All',
                date: 'All',
                status: 'All'
            }
        };
    },
    computed: {
        loadedSubmissions() {
            let loadedData = this.tableData;
            let filteredData = filterData(loadedData, this.appliedFilters);
            return filteredData.slice(0, this.rowsToLoad * this.timesLoaded);
        },
        showButton() {
            return this.rowsToLoad * this.timesLoaded <= this.loadedSubmissions.length;
        }
    },
    methods: {
        loadMore() {
            this.timesLoaded++;
        },
        postsPerPage(e) {
            this.timesLoaded = 1;
            this.rowsToLoad = e.target.value;
        },
        toggleFilter() {
            this.filterModal = !this.filterModal;
        },
        stockFilter(data) {
            this.selectedFilters = data;
        },
        submitFilter() {
            this.appliedFilters.form = this.selectedFilters.form;
            this.appliedFilters.status = this.selectedFilters.status;
            this.appliedFilters.recommendation = this.selectedFilters.recommendation;
            this.appliedFilters.date = this.selectedFilters.date;
            this.appliedFilters.form_type = this.selectedFilters.form_type;
            this.toggleFilter();
        },
        triggerModal(passedData) {
            this.modalData = passedData;
            this.modalActive = !this.modalActive;
        },
        deleteSubmission(item) {
            let deleteIndex = this.tableData.findIndex(x => x == item);
            this.$emit('delete-this-submission', deleteIndex);
        },
        editSubmission(item) {
            this.$emit('edit-this-submission', item);
        },
        createNewSubmission() {
            this.$emit('create-new-submission');
        }
    }
};
</script>

<style>
.margin-left {
    margin-left: 10px;
}
.row-hover:hover {
    box-shadow: 2px 2px 4px 4px rgba(0, 0, 0, 0.2);
    transform: scale(1.001);
}
</style>
