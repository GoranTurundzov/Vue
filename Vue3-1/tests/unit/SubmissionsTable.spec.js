import { mount } from '@vue/test-utils';
import SubmissionsTable from '@/components/SubmissionsTable';
import SubmissionsTableRow from '@/components/SubmissionsTableRow';
import dummyData from '../../src/assets/DummyData';
import Modal from '../../src/components/Modal';
import formJson from '../../public/forms.json';
import SubmissionTableFilter from '../../src/components/SubmissionTableFilter';

let wrapper;
let spySubmitFilterFn = jest.spyOn(SubmissionsTable.methods, 'submitFilter');
let spyToggleFilterFn = jest.spyOn(SubmissionsTable.methods, 'toggleFilter');
let spyLoadMoreFn = jest.spyOn(SubmissionsTable.methods, 'loadMore');
let spyPostsPerPageFn = jest.spyOn(SubmissionsTable.methods, 'postsPerPage');
let spyTriggerModalFn = jest.spyOn(SubmissionsTable.methods, 'triggerModal');
let spyStockFilterFn = jest.spyOn(SubmissionsTable.methods, 'stockFilter');
let spyDeleteSubmissionFn = jest.spyOn(SubmissionsTable.methods, 'deleteSubmission');

describe('SubmissionsTable', () => {
    beforeEach(() => {
        wrapper = mount(SubmissionsTable, {
            propsData: {
                tableData: dummyData,
                formSelectOptions: formJson
            }
        });
    });

    it('Matches snapshot', () => expect(wrapper.element).toMatchSnapshot());

    it('Renders the right number of rows and button dissappears', async () => {
        let submissionsRows = wrapper.findAllComponents(SubmissionsTableRow);

        expect(submissionsRows).toHaveLength(wrapper.vm.rowsToLoad * wrapper.vm.timesLoaded);

        let button = wrapper.find('[data-testid="test-load-more"]');

        expect(button.exists()).toBeTruthy();
        expect(wrapper.vm.timesLoaded).toEqual(1);

        await button.trigger('click');
        expect(spyLoadMoreFn).toBeCalled();
        expect(wrapper.vm.timesLoaded).toEqual(2);
        expect(wrapper.vm.showButton).toBeTruthy();

        await button.trigger('click');
        expect(spyLoadMoreFn).toBeCalled();
        expect(wrapper.vm.showButton).toBeTruthy();
        expect(wrapper.vm.timesLoaded).toEqual(3);

        await button.trigger('click');
        expect(spyLoadMoreFn).toBeCalled();
        expect(wrapper.vm.showButton).toBeTruthy();
        expect(wrapper.vm.timesLoaded).toEqual(4);

        await button.trigger('click');
        expect(spyLoadMoreFn).toBeCalled();
        expect(wrapper.vm.timesLoaded).toEqual(5);
        expect(wrapper.vm.showButton).toBeFalsy();
    });

    test('It Renders the Table rows', () => {
        expect(wrapper.findComponent(SubmissionsTableRow).exists()).toBe(true);
    });

    test('Number of posts change when option is changed', async () => {
        await wrapper.findAll('option')[1].setSelected();
        expect(spyPostsPerPageFn).toBeCalled();
        expect(wrapper.findAllComponents(SubmissionsTableRow)).toHaveLength(
            parseInt(wrapper.vm.rowsToLoad)
        );

        await wrapper.find('[data-testid="test-load-more"]').trigger('click');
        expect(spyLoadMoreFn).toBeCalled();
        expect(wrapper.findAllComponents(SubmissionsTableRow)).toHaveLength(
            wrapper.vm.rowsToLoad * 2
        );
    });

    test('Loads the modal with data', async () => {
        expect(wrapper.vm.modalActive).toBe(false);
        expect(wrapper.vm.modalData).toBeNull();
        expect(wrapper.findComponent(Modal).exists()).toBe(false);

        let component = wrapper.findAllComponents(SubmissionsTableRow)[2];

        await component.find('[data-testid="modal-activator"]').trigger('click');
        expect(spyTriggerModalFn).toBeCalled();
        expect(wrapper.vm.modalActive).toBe(true);
        expect(wrapper.vm.modalData).not.toBeNull();
        expect(wrapper.findComponent(Modal).exists()).toBe(true);

        let modalWrapper = mount(Modal, {
            slots: {
                default: '<div>' + wrapper.vm.modalData.form_type + '</div>'
            }
        });
        expect(modalWrapper.find('.block-content').html()).toContain('<div>page</div>');
    });

    test('Filter Data Works', async () => {
        let testFilters = {
            form: 'Form 1',
            form_type: 'All',
            recommendation: 'Yes',
            date: 'Month to date',
            status: 'pending'
        };

        await wrapper.setData({
            appliedFilters: testFilters
        });

        let filteredObjectToBeFound = {
            name: 'Ivana Trajkova',
            email: 'ivana_tr@example.com',
            recommendation: 'yes',
            feedback: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Maecenas non enim gravida ante feugiat fermentum in vitae sem. Mauris ligula ante, finibus quis convallis eget, auctor vehicula massa. Nunc arcu lectus, viverra ac enim eu, suscipit fermentum sapien. Nam maximus facilisis enim sit amet tempor. Curabitur at euismod augue, et aliquet elit. Mauris dui libero, consequat sit amet lacus sit amet, vulputate interdum tellus. Morbi condimentum porttitor ornare.',
            date: '03/10/2021',
            form: 'Form 1',
            form_type: 'popup',
            status: 'pending'
        };
        expect(wrapper.vm.loadedSubmissions[0]).toEqual(filteredObjectToBeFound);
    });

    test('on filter change it stocks filters', async () => {
        await wrapper.setData({
            filterModal: true
        });
        await wrapper.findComponent(SubmissionTableFilter).trigger('change-filter');
        expect(spyStockFilterFn).toBeCalled();
    });

    test('Activates filter(toggleFilter) modal and submits filters', async () => {
        wrapper.setData({
            appliedFilters: {
                form: 'All',
                form_type: 'All',
                recommendation: 'All',
                date: 'All',
                status: 'All'
            },
            selectedFilters: {
                form_type: 'Email',
                recommendation: 'All',
                date: 'Today',
                status: 'All',
                form: 'All'
            }
        });
        expect(wrapper.vm.filterModal).toBe(false);

        await wrapper.find('[data-testid="open-filters-button"]').trigger('click');
        expect(spyToggleFilterFn).toBeCalled();
        expect(wrapper.vm.filterModal).toBe(true);

        await wrapper.find('[data-testid="submit-filters-button"]').trigger('click');
        expect(spySubmitFilterFn).toBeCalled();
        expect(wrapper.vm.appliedFilters).toEqual(wrapper.vm.selectedFilters);
    });

    test('submission-table-filter recieves correct values as props', async () => {
        wrapper.setData({
            appliedFilters: {
                form: 'All',
                form_type: 'Email',
                recommendation: 'Maybe',
                date: 'All',
                status: 'Approved'
            }
        });

        await wrapper.find('[data-testid="open-filters-button"]').trigger('click');

        expect(spyToggleFilterFn).toBeCalled();
        expect(wrapper.findComponent(SubmissionTableFilter).exists()).toBe(true);
        expect(wrapper.findComponent(SubmissionTableFilter).vm.form).toEqual('All');
        expect(wrapper.findComponent(SubmissionTableFilter).vm.form_type).toEqual('Email');
        expect(wrapper.findComponent(SubmissionTableFilter).vm.recommendation).toEqual('Maybe');
        expect(wrapper.findComponent(SubmissionTableFilter).vm.date).toEqual('All');
        expect(wrapper.findComponent(SubmissionTableFilter).vm.status).toEqual('Approved');
    });

    it('Listenes to the event emmited by the row component and emmits new event', () => {
        let randomRowComponent = wrapper.findAllComponents(SubmissionsTableRow)[3];
        randomRowComponent.find('[ data-testid="delete-activator"]').trigger('click');
        expect(spyDeleteSubmissionFn).toBeCalled();
        expect(wrapper.emitted('delete-this-submission')).toBeTruthy();
        expect(wrapper.emitted('delete-this-submission')).toHaveLength(1);
        expect(wrapper.emitted('delete-this-submission')[0][0]).toEqual(3);
    });

    test('on filter change it stocks filters', async () => {
        await wrapper.setData({
            filterModal: true
        });
        await wrapper.findComponent(SubmissionTableFilter).trigger('change-filter');
        expect(spyStockFilterFn).toBeCalled();
    });
});
