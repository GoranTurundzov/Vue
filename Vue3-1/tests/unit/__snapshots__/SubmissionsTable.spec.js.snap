// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubmissionsTable Matches snapshot 1`] = `
<div
  data-v-app=""
>
  
  <div
    class="block-content"
  >
    <div
      class="block"
    >
      <select
        class="btn btn-default dropdown-toggle"
      >
        <option>
          5
        </option>
        <option>
          10
        </option>
        <option>
          15
        </option>
        <option>
          20
        </option>
      </select>
      <button
        class="btn btn-secondary margin-left"
        data-testid="open-filters-button"
      >
         Filter submissions 
      </button>
      <button
        class="btn btn-primary pull-right"
        data-testid="create-submission-button"
      >
         Create submission 
      </button>
    </div>
    <div
      class="block block-bordered block-rounded"
    >
      
      <div
        class="content justify-center row-hover"
      >
        
        <div
          class="column"
        >
          <div
            class="title text-capitalize"
          >
            name
          </div>
          <div
            class="font-w600 data-content name"
          >
            Ivana Trajkova
          </div>
        </div>
        <div
          class="column"
        >
          <div
            class="title text-capitalize"
          >
            email
          </div>
          <div
            class="font-w600 data-content email"
          >
            ivana_tr@example.com
          </div>
        </div>
        <div
          class="column"
        >
          <div
            class="title text-capitalize"
          >
            recommendation
          </div>
          <div
            class="font-w600 data-content recommendation"
          >
            yes
          </div>
        </div>
        <div
          class="column"
        >
          <div
            class="title text-capitalize"
          >
            feedback
          </div>
          <div
            class="font-w600 data-content feedback"
          >
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Maecenas non enim gravida ante feugiat fermentum in vitae sem. Mauris ligula ante, finibus quis convallis eget, auctor vehicula massa. Nunc arcu lectus, viverra ac enim eu, suscipit fermentum sapien. Nam maximus facilisis enim sit amet tempor. Curabitur at euismod augue, et aliquet elit. Mauris dui libero, consequat sit amet lacus sit amet, vulputate interdum tellus. Morbi condimentum porttitor ornare.
          </div>
        </div>
        <div
          class="column"
        >
          <div
            class="title text-capitalize"
          >
            date
          </div>
          <div
            class="font-w600 data-content date"
          >
            03/10/2021
          </div>
        </div>
        <div
          class="column"
        >
          <div
            class="title text-capitalize"
          >
            form
          </div>
          <div
            class="font-w600 data-content form"
          >
            Form 1
          </div>
        </div>
        <div
          class="column"
        >
          <div
            class="title text-capitalize"
          >
            form type
          </div>
          <div
            class="font-w600 data-content form_type"
          >
            popup
          </div>
        </div>
        <div
          class="column"
        >
          <div
            class="title text-capitalize"
          >
            status
          </div>
          <div
            class="font-w600 data-content status"
          >
            pending
          </div>
        </div>
        
        <div
          class="column"
        >
          <div>
            Action
          </div>
          <div>
            <button
              class="btn btn-gray btn-sm push-5-r"
            >
              <i
                class="fa fa-edit"
              />
            </button>
            <button
              class="btn btn-gray btn-sm push-5-r"
              data-testid="modal-activator"
            >
              <i
                class="fa fa-eye"
              />
            </button>
            <button
              class="btn btn-gray btn-sm push-5-r"
              data-testid="delete-activator"
            >
              <i
                class="fa fa-trash"
              />
            </button>
          </div>
        </div>
      </div>
      <div
        class="content justify-center row-hover"
      >
        
        <div
          class="column"
        >
          <div
            class="title text-capitalize"
          >
            name
          </div>
          <div
            class="font-w600 data-content name"
          >
            Ivona Atanasova
          </div>
        </div>
        <div
          class="column"
        >
          <div
            class="title text-capitalize"
          >
            email
          </div>
          <div
            class="font-w600 data-content email"
          >
            ivana_ar@example.com
          </div>
        </div>
        <div
          class="column"
        >
          <div
            class="title text-capitalize"
          >
            recommendation
          </div>
          <div
            class="font-w600 data-content recommendation"
          >
            maybe
          </div>
        </div>
        <div
          class="column"
        >
          <div
            class="title text-capitalize"
          >
            feedback
          </div>
          <div
            class="font-w600 data-content feedback"
          >
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Maecenas non enim gravida ante feugiat fermentum in vitae sem. Mauris ligula ante, finibus quis conva
          </div>
        </div>
        <div
          class="column"
        >
          <div
            class="title text-capitalize"
          >
            date
          </div>
          <div
            class="font-w600 data-content date"
          >
            10/11/2020
          </div>
        </div>
        <div
          class="column"
        >
          <div
            class="title text-capitalize"
          >
            form
          </div>
          <div
            class="font-w600 data-content form"
          >
            Form 2
          </div>
        </div>
        <div
          class="column"
        >
          <div
            class="title text-capitalize"
          >
            form type
          </div>
          <div
            class="font-w600 data-content form_type"
          >
            page
          </div>
        </div>
        <div
          class="column"
        >
          <div
            class="title text-capitalize"
          >
            status
          </div>
          <div
            class="font-w600 data-content status"
          >
            approved
          </div>
        </div>
        
        <div
          class="column"
        >
          <div>
            Action
          </div>
          <div>
            <button
              class="btn btn-gray btn-sm push-5-r"
            >
              <i
                class="fa fa-edit"
              />
            </button>
            <button
              class="btn btn-gray btn-sm push-5-r"
              data-testid="modal-activator"
            >
              <i
                class="fa fa-eye"
              />
            </button>
            <button
              class="btn btn-gray btn-sm push-5-r"
              data-testid="delete-activator"
            >
              <i
                class="fa fa-trash"
              />
            </button>
          </div>
        </div>
      </div>
      <div
        class="content justify-center row-hover"
      >
        
        <div
          class="column"
        >
          <div
            class="title text-capitalize"
          >
            name
          </div>
          <div
            class="font-w600 data-content name"
          >
            Jonathan
          </div>
        </div>
        <div
          class="column"
        >
          <div
            class="title text-capitalize"
          >
            email
          </div>
          <div
            class="font-w600 data-content email"
          >
            jonathan@gmail.com
          </div>
        </div>
        <div
          class="column"
        >
          <div
            class="title text-capitalize"
          >
            recommendation
          </div>
          <div
            class="font-w600 data-content recommendation"
          >
            maybe
          </div>
        </div>
        <div
          class="column"
        >
          <div
            class="title text-capitalize"
          >
            feedback
          </div>
          <div
            class="font-w600 data-content feedback"
          >
            My neighbor Karly has one of these. She works as a gambler and she says it looks tall. I saw one of these in Spratly Islands and I bought one. works okay.
          </div>
        </div>
        <div
          class="column"
        >
          <div
            class="title text-capitalize"
          >
            date
          </div>
          <div
            class="font-w600 data-content date"
          >
            16/03/2021
          </div>
        </div>
        <div
          class="column"
        >
          <div
            class="title text-capitalize"
          >
            form
          </div>
          <div
            class="font-w600 data-content form"
          >
            Form 3
          </div>
        </div>
        <div
          class="column"
        >
          <div
            class="title text-capitalize"
          >
            form type
          </div>
          <div
            class="font-w600 data-content form_type"
          >
            page
          </div>
        </div>
        <div
          class="column"
        >
          <div
            class="title text-capitalize"
          >
            status
          </div>
          <div
            class="font-w600 data-content status"
          >
            approved
          </div>
        </div>
        
        <div
          class="column"
        >
          <div>
            Action
          </div>
          <div>
            <button
              class="btn btn-gray btn-sm push-5-r"
            >
              <i
                class="fa fa-edit"
              />
            </button>
            <button
              class="btn btn-gray btn-sm push-5-r"
              data-testid="modal-activator"
            >
              <i
                class="fa fa-eye"
              />
            </button>
            <button
              class="btn btn-gray btn-sm push-5-r"
              data-testid="delete-activator"
            >
              <i
                class="fa fa-trash"
              />
            </button>
          </div>
        </div>
      </div>
      <div
        class="content justify-center row-hover"
      >
        
        <div
          class="column"
        >
          <div
            class="title text-capitalize"
          >
            name
          </div>
          <div
            class="font-w600 data-content name"
          >
            Nicole
          </div>
        </div>
        <div
          class="column"
        >
          <div
            class="title text-capitalize"
          >
            email
          </div>
          <div
            class="font-w600 data-content email"
          >
            nicole@yahoo.com
          </div>
        </div>
        <div
          class="column"
        >
          <div
            class="title text-capitalize"
          >
            recommendation
          </div>
          <div
            class="font-w600 data-content recommendation"
          >
            no
          </div>
        </div>
        <div
          class="column"
        >
          <div
            class="title text-capitalize"
          >
            feedback
          </div>
          <div
            class="font-w600 data-content feedback"
          >
            The box this comes in is 4 yard by 5 inch and weights 12 pound! i use it profusely when i'm in my garage.
          </div>
        </div>
        <div
          class="column"
        >
          <div
            class="title text-capitalize"
          >
            date
          </div>
          <div
            class="font-w600 data-content date"
          >
            02/10/2021
          </div>
        </div>
        <div
          class="column"
        >
          <div
            class="title text-capitalize"
          >
            form
          </div>
          <div
            class="font-w600 data-content form"
          >
            Form 4
          </div>
        </div>
        <div
          class="column"
        >
          <div
            class="title text-capitalize"
          >
            form type
          </div>
          <div
            class="font-w600 data-content form_type"
          >
            popup
          </div>
        </div>
        <div
          class="column"
        >
          <div
            class="title text-capitalize"
          >
            status
          </div>
          <div
            class="font-w600 data-content status"
          >
            pending
          </div>
        </div>
        
        <div
          class="column"
        >
          <div>
            Action
          </div>
          <div>
            <button
              class="btn btn-gray btn-sm push-5-r"
            >
              <i
                class="fa fa-edit"
              />
            </button>
            <button
              class="btn btn-gray btn-sm push-5-r"
              data-testid="modal-activator"
            >
              <i
                class="fa fa-eye"
              />
            </button>
            <button
              class="btn btn-gray btn-sm push-5-r"
              data-testid="delete-activator"
            >
              <i
                class="fa fa-trash"
              />
            </button>
          </div>
        </div>
      </div>
      <div
        class="content justify-center row-hover"
      >
        
        <div
          class="column"
        >
          <div
            class="title text-capitalize"
          >
            name
          </div>
          <div
            class="font-w600 data-content name"
          >
            Ivana Trajkova
          </div>
        </div>
        <div
          class="column"
        >
          <div
            class="title text-capitalize"
          >
            email
          </div>
          <div
            class="font-w600 data-content email"
          >
            ivana_tr@example.com
          </div>
        </div>
        <div
          class="column"
        >
          <div
            class="title text-capitalize"
          >
            recommendation
          </div>
          <div
            class="font-w600 data-content recommendation"
          >
            yes
          </div>
        </div>
        <div
          class="column"
        >
          <div
            class="title text-capitalize"
          >
            feedback
          </div>
          <div
            class="font-w600 data-content feedback"
          >
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Maecenas non enim gravida ante feugiat fermentum in vitae sem. Mauris ligula ante, finibus quis convallis eget, auctor vehicula massa. Nunc arcu lectus, viverra ac enim eu, suscipit fermentum sapien. Nam maximus facilisis enim sit amet tempor. Curabitur at euismod augue, et aliquet elit. Mauris dui libero, consequat sit amet lacus sit amet, vulputate interdum tellus. Morbi condimentum porttitor ornare.
          </div>
        </div>
        <div
          class="column"
        >
          <div
            class="title text-capitalize"
          >
            date
          </div>
          <div
            class="font-w600 data-content date"
          >
            13/04/2021
          </div>
        </div>
        <div
          class="column"
        >
          <div
            class="title text-capitalize"
          >
            form
          </div>
          <div
            class="font-w600 data-content form"
          >
            Form 1
          </div>
        </div>
        <div
          class="column"
        >
          <div
            class="title text-capitalize"
          >
            form type
          </div>
          <div
            class="font-w600 data-content form_type"
          >
            popup
          </div>
        </div>
        <div
          class="column"
        >
          <div
            class="title text-capitalize"
          >
            status
          </div>
          <div
            class="font-w600 data-content status"
          >
            pending
          </div>
        </div>
        
        <div
          class="column"
        >
          <div>
            Action
          </div>
          <div>
            <button
              class="btn btn-gray btn-sm push-5-r"
            >
              <i
                class="fa fa-edit"
              />
            </button>
            <button
              class="btn btn-gray btn-sm push-5-r"
              data-testid="modal-activator"
            >
              <i
                class="fa fa-eye"
              />
            </button>
            <button
              class="btn btn-gray btn-sm push-5-r"
              data-testid="delete-activator"
            >
              <i
                class="fa fa-trash"
              />
            </button>
          </div>
        </div>
      </div>
      
    </div>
    <div
      class="row justify-content-center"
    >
      <button
        class="btn btn-primary justify-content-center"
        data-testid="test-load-more"
      >
         Load More 
      </button>
    </div>
  </div>
  <!--v-if-->
  <!--v-if-->
  
</div>
`;
